mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go.erb

```diff
@@ -1832,18 +1832,6 @@ func resourceBigQueryTableDelete(d *schema.ResourceData, meta interface{}) error
 	if d.Get("deletion_protection").(bool) {
 		return fmt.Errorf("cannot destroy table %v without setting deletion_protection=false and running `terraform apply`", d.Id())
 	}
-	<% unless version == 'ga' -%>
-	if v, ok := d.GetOk("resource_tags"); ok {
```

Comment:
I think this is a breaking change that should be avoided a minor release. This is `Changing resource deletion behavior` in https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes/#resource-level-breaking-changes.

We have a couple of options:

1. wait until the next major release, which is expected in a couple of months
2. introduce a virtual field to force delete the resource even when resource_tags are present and default it to false.  

Please let me know your thoughts and if you have any questions. Thanks!

Reply:
Would the combination of the points below make it eligible for a minor version release?

- Feature was very newly added (released on 4/22)
- Feature is in the beta provider
- The new deletion behavior is more permissive instead of more restrictive

Reply:
@wj-chen Thanks for the quick response.

> * Feature was very newly added (released on 4/22)
> * Feature is in the beta provider

The beta provider is also a stable surface that does not allow breaking changes outside of major releases. See go/terraform-launch-stages#google-beta

> * The new deletion behavior is more permissive instead of more restrictive

I'd still think it's breaking since it's changing the default deletion behavior and I'm a bit hesitant about changing it directly in a minor. However, if an urgent fix is needed in a minor, we could follow https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/make-a-breaking-change/#in-minor-releases - "this change can be made if the default behavior stays the same and new behavior can be enabled with a flag", such as adding a `force_delete` field, which is the option 2 I proposed in https://github.com/GoogleCloudPlatform/magic-modules/pull/10568#discussion_r1588026232. Please let me know what your thought on this approach and if it will work for your use case. Thanks!

Reply:
Will discuss internally regarding the migration path for the next major release

Reply:
Hi @shuyama1, I'd like to follow up on this discussion again since the submission window for the next major release is now open. We want to promote this field to the GA provider, and make the default value `true` in the GA provider from the get-go if possible.

* Would it count as a breaking change if the promotion and default value update take place together? If not, we'll proceed with that since it's the most ideal. Updating the default value for the beta provider will be treated as a separate breaking change.
* Otherwise, we would most likely promote first then make a breaking change to update the default value for both providers as a follow-up. This way the GA promotion won't be blocked by the next major release timeline.

Reply:
@wj-chen Thank you for following up on this issue.

Our policy requires that the beta provider be a superset of the GA provider. This means the features and behaviors in GA should always be consistent with those in beta.

If we want to promote the feature to GA before changing the default in beta in 6.0.0 (as you correctly noted, this default value swap is a breaking change and must wait until the 6.0.0 major release), we need to use the current default value and update the default values for both GA and beta later in the 6.0.0 release.

Hope this answers your question and let me know if you have further questions.

Reply:
In this case we'll proceed with promoting the feature to GA first, then making a breaking change to change the default value for the field for both GA and beta providers in the 6.0.0 release. Thank you for the guidance!

Reply:
Perfect. Sounds good!